File: Useful/bizness.py
#!/usr/bin/env python3

import hashlib
import base64
import os

hash_type = "SHA1"
salt = "d"
search = "$SHA1$d$uP0_QaVBpDWFeo8-dRzDqRwXQ2I="
wordlist = './rockyou.txt'

def hash_password(hash_type, salt, password):
    hash_obj = hashlib.new(hash_type)
    hash_obj.update(salt.encode('utf-8'))
    hash_obj.update(password)
    hashed_bytes = hash_obj.digest()
    return f"${hash_type}${salt}${base64.urlsafe_b64encode(hashed_bytes).decode('utf-8').replace('+', '.')}"

def search_password(search, wordlist):
    with open(wordlist, 'r', encoding='latin-1') as password_list:
        for password in password_list:
            hashed_password = hash_password("SHA1", "d", password.strip().encode('utf-8'))
            if hashed_password == search:
                print(f'Found Password: {password.strip()}, hash: {hashed_password}')

search_password(search, wordlist)

File: Useful/bizness.go
package main

import (
	"crypto/sha1"
	"encoding/base64"
	"fmt"
	"io/ioutil"
	"strings"
)

func hashPassword(salt string, password []byte) string {
	hash := sha1.New()
	hash.Write([]byte(salt))
	hash.Write(password)
	hashedBytes := hash.Sum(nil)
	return fmt.Sprintf("$SHA1$%s$%s", salt, base64.URLEncoding.EncodeToString(hashedBytes))
}

func searchPassword(search, wordlistPath string) {
	wordlist, err := ioutil.ReadFile(wordlistPath)
	if err != nil {
		fmt.Println("Error reading wordlist:", err)
		return
	}

	lines := strings.Split(string(wordlist), "\n")
	for _, password := range lines {
		password = strings.TrimSpace(password) // Supprimer les espaces autour du mot de passe

		if password == "" {
			continue // Ignorer les lignes vides
		}

		hashedPassword := hashPassword("d", []byte(password))

		if hashedPassword == search {
			fmt.Printf("Found Password: %s, hash: %s\n", password, hashedPassword)
			return
		}
	}
	fmt.Println("Password not found.")
}

func main() {
	search := "$SHA1$d$uP0_QaVBpDWFeo8-dRzDqRwXQ2I="
	wordlistPath := "./rockyou.txt"

	searchPassword(search, wordlistPath)
}
